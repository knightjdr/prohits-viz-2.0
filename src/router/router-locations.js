import PropTypes from 'prop-types';
import React, { Fragment } from 'react';

import { Route, Switch, withRouter } from 'react-router-dom';

// routes
import Analysis from '../analysis/analysis-loadable';
import App from '../App';
import Help from '../help/help-loadable';
import Missing from '../404/404';
import Navbar from '../navbar/navbar-container';
import News from '../news/news';
import Tasks from '../tasks/tasks-container';
import Visualization from '../visualization/visualization-loadable';

export const Locations = ({ location }) => (
  <Fragment>
    <Navbar
      background={location.pathname === '/' ? 'semi' : 'dark'}
      fixed={location.pathname.startsWith('/help')}
    />
    <Switch>
      <Route
        exact
        path="/"
        component={App}
      />
      <Route
        path="/analysis"
        component={Analysis}
      />
      <Route
        path="/help"
        component={Help}
      />
      <Route
        path="/news"
        component={News}
      />
      <Route
        path="/tasks"
        component={Tasks}
      />
      <Route
        path="/visualization"
        component={Visualization}
      />
      <Route
        path="*"
        component={Missing}
      />
    </Switch>
  </Fragment>
);

Locations.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string,
  }).isRequired,
};

export default withRouter(Locations);
