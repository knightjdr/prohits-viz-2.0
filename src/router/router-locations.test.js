import React from 'react';
import { shallow } from 'enzyme';

import { Locations } from './router-locations';

describe('Router locations', () => {
  describe('home route', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <Locations
          location={{ pathname: '/' }}
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have navbar with semi background', () => {
      expect(wrapper.find('Connect(NavbarContainer)').props().background).toBe('semi');
    });
  });

  describe('help route', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <Locations
          location={{ pathname: '/help/introduction' }}
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have a fixed navbar', () => {
      expect(wrapper.find('Connect(NavbarContainer)').props().fixed).toBeTruthy();
    });
  });

  describe('other route', () => {
    let wrapper;

    beforeAll(() => {
      wrapper = shallow(
        <Locations
          location={{ pathname: '/news' }}
        />,
      );
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have navbar with dark background', () => {
      expect(wrapper.find('Connect(NavbarContainer)').props().background).toBe('dark');
    });

    it('should not have a fixed navbar', () => {
      expect(wrapper.find('Connect(NavbarContainer)').props().fixed).toBeFalsy();
    });
  });
});
