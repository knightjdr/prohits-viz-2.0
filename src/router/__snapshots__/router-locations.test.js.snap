// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Router locations help route should match snapshot 1`] = `
<React.Fragment>
  <Connect(NavbarContainer)
    background="dark"
    fixed={true}
  />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/analysis"
    />
    <Route
      component={[Function]}
      path="/help"
    />
    <Route
      component={[Function]}
      path="/news"
    />
    <Route
      component={[Function]}
      path="/tasks"
    />
    <Route
      component={[Function]}
      path="/visualization"
    />
    <Route
      component={[Function]}
      path="*"
    />
  </Switch>
</React.Fragment>
`;

exports[`Router locations home route should match snapshot 1`] = `
<React.Fragment>
  <Connect(NavbarContainer)
    background="semi"
    fixed={false}
  />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/analysis"
    />
    <Route
      component={[Function]}
      path="/help"
    />
    <Route
      component={[Function]}
      path="/news"
    />
    <Route
      component={[Function]}
      path="/tasks"
    />
    <Route
      component={[Function]}
      path="/visualization"
    />
    <Route
      component={[Function]}
      path="*"
    />
  </Switch>
</React.Fragment>
`;

exports[`Router locations other route should match snapshot 1`] = `
<React.Fragment>
  <Connect(NavbarContainer)
    background="dark"
    fixed={false}
  />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/analysis"
    />
    <Route
      component={[Function]}
      path="/help"
    />
    <Route
      component={[Function]}
      path="/news"
    />
    <Route
      component={[Function]}
      path="/tasks"
    />
    <Route
      component={[Function]}
      path="/visualization"
    />
    <Route
      component={[Function]}
      path="*"
    />
  </Switch>
</React.Fragment>
`;
